staging/kos/pkg/controllers/subnet/validator.go

```diff
@@ -156,8 +152,11 @@ func (v *Validator) OnUpdate(oldObj, newObj interface{}) {
 	oldS, newS := oldObj.(*netv1a1.Subnet), newObj.(*netv1a1.Subnet)
 	glog.V(5).Infof("Notified of update from %#+v to %#+v.", oldS, newS)
 
-	// Process a subnet only if the fields that affect validation have changed.
-	if oldS.Spec.IPv4 != newS.Spec.IPv4 || oldS.Spec.VNI != newS.Spec.VNI {
+	// Subnet updates convey nothing useful validation-wise, because only VNI
+	// and CIDR block affect validation and they're immutable. But a subnet
+	// could still be deleted and re-created with different VNI and/or CIDR
+	// block, and the two events could be delivered as an update notification.
+	if oldS.UID != newS.UID {
```

Comment:
The fields that matter towards validation can only change if UID changes: enqueue only if that's the case.